src/__tests__/core/account.storage.test.ts

```diff
@@ -0,0 +1,243 @@
+            value: '0x776f726c6400000000000000000000000000000000000000000000000000000a',
+          },
+        },
+      },
+    },
+  };
+
+  const result = await execQuery(query);
+  expect(result).toEqual(expected);
+});
+
+test('account->storage: retrieve map storage with numerical key', async () => {
+  const query = `
+    {
+      account(address: "0x171AEEfFa86Fbfd8c9d25A235f1dDBB03A635742") {
+        storage {
+          solidityMap(at: 8, keyType: number) {
+            value(at: "1234")
+          }
+        }
+      }
+    }
+    `;
+
+  const expected = {
+    data: {
+      account: {
+        storage: {
+          solidityMap: {
+            value: '0x00000000000000000000000000000000000000000000000000000000000010e1',
+          },
+        },
+      },
+    },
+  };
+
+  const result = await execQuery(query);
+  expect(result).toEqual(expected);
+});
+
+test('account->storage: retrieve nested map storage inside a fixed list', async () => {
+  const query = `
+    {
+      account(address: "0x171AEEfFa86Fbfd8c9d25A235f1dDBB03A635742") {
+        storage {
+          solidityFixedArray(at: 9) {
+            solidityMap(at: 11, keyType: string) {
+              value(at: "marco")
+            }
+          }
+        }
+      }
+    }
+    `;
+
+  const expected = {
+    data: {
+      account: {
+        storage: {
+          solidityFixedArray: {
+            solidityMap: {
+              value: '0x0000000000000000000000000000000000000000000000000000000000001b62',
+            },
+          },
+        },
+      },
+    },
+  };
+
+  const result = await execQuery(query);
+  expect(result).toEqual(expected);
+});
+
+test('account->storage: retrieve nested dynamic list storage inside a map', async () => {
+  const query = `
+    {
+      account(address: "0x171AEEfFa86Fbfd8c9d25A235f1dDBB03A635742") {
+        storage {
+          solidityMap(at: 29, keyType: address) {
+            solidityDynamicArray(at: "0x874B54A8bD152966d63F706BAE1FfeB0411921E5") {
+              value(at: 10)
+            }
+          }
+        }
+      }
+    }
+    `;
+
+  const expected = {
+    data: {
+      account: {
+        storage: {
+          solidityMap: {
+            solidityDynamicArray: {
+              value: '0x6920616d206e6f7420706f736974696f6e20313530000000000000000000002a',
+            },
+          },
+        },
+      },
+    },
+  };
+
+  const result = await execQuery(query);
+  expect(result).toEqual(expected);
+});
+
+test('account->storage: retrieve nested dynamic list storage inside a fixed list', async () => {
+  const query = `
+    {
+      account(address: "0x171AEEfFa86Fbfd8c9d25A235f1dDBB03A635742") {
+        storage {
+          solidityFixedArray(at: 30) {
+            solidityDynamicArray(at: 40) {
+              value(at: 0)
+            }
+          }
+        }
+      }
+    }
+    `;
+
+  const expected = {
+    data: {
+      account: {
+        storage: {
+          solidityFixedArray: {
```

Comment:
I was experimenting with a similar solution earlier and it felt rather cluttered with keyType. I agree that having keyType as an attribute is a good idea for future expansion but because there will only ever be two types of arrays, I think this is a better solution. If you feel otherwise though I'd be happy to implement it!

Reply:
Ok, you do make a valid point from viewpoint of UX/DX. LGTM.

Reply:
SGTM as well :)